engine/runtime-integration-tests/src/test/java/org/enso/compiler/ImportsAndFQNConsistencyTest.java

```diff
@@ -0,0 +1,152 @@
+package org.enso.compiler;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.enso.interpreter.test.TestBase;
+import org.graalvm.polyglot.Context;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.experimental.theories.DataPoints;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runner.RunWith;
+import org.junit.runners.model.Statement;
+
+/**
+ * If a symbol is importable by an import statement, then it should also be accessible via FQN
+ * without any import, or with an import of its parent module. This rule is enforced by this test.
+ * It uses symbols from standard libraries.
+ */
+@RunWith(Theories.class)
+public class ImportsAndFQNConsistencyTest extends TestBase {
+  private static Context ctx;
+
+  /** Used for description in {@link PrintCodeRule} test rule. */
+  private static String code;
+
+  @Rule public final TestRule printCodeRule = new PrintCodeRule();
+
+  @DataPoints
+  public static final List<Symbol> symbols =
+      List.of(new Symbol("Standard.Base.Data.Vector"), new Symbol("Standard.Table.Data.Table"));
```

Comment:
You could read all the exported symbols from [Main.enso](https://github.com/enso-org/enso/blob/a01aeab3a4acacf8f6a77ba94097f4cca16c8bc0/distribution/lib/Standard/Base/0.0.0-dev/src/Main.enso#L101) of `Standard.Base` and also other libraries.

Reply:
Good point. The main idea is presented in 9e1252626dcd222934a2ee116b6dfca404360747